presto-hive/src/main/java/io/prestosql/plugin/hive/procedure/CreateEmptyPartitionProcedure.java

```diff
@@ -94,21 +97,28 @@ public Procedure get()
                 CREATE_EMPTY_PARTITION.bindTo(this));
     }
 
-    public void createEmptyPartition(ConnectorSession session, String schema, String table, List<String> partitionColumnNames, List<String> partitionValues)
+    public void createEmptyPartition(ConnectorSession session, ConnectorAccessControl accessControl, String schema, String table, List<String> partitionColumnNames, List<String> partitionValues)
     {
         try (ThreadContextClassLoader ignored = new ThreadContextClassLoader(getClass().getClassLoader())) {
-            doCreateEmptyPartition(session, schema, table, partitionColumnNames, partitionValues);
+            doCreateEmptyPartition(session, accessControl, schema, table, partitionColumnNames, partitionValues);
         }
     }
 
-    private void doCreateEmptyPartition(ConnectorSession session, String schemaName, String tableName, List<String> partitionColumnNames, List<String> partitionValues)
+    private void doCreateEmptyPartition(ConnectorSession session, ConnectorAccessControl accessControl, String schemaName, String tableName, List<String> partitionColumnNames, List<String> partitionValues)
     {
         TransactionalMetadata hiveMetadata = hiveMetadataFactory.create();
         HiveTableHandle tableHandle = (HiveTableHandle) hiveMetadata.getTableHandle(session, new SchemaTableName(schemaName, tableName));
         if (tableHandle == null) {
             throw new PrestoException(INVALID_PROCEDURE_ARGUMENT, format("Table '%s' does not exist", new SchemaTableName(schemaName, tableName)));
         }
 
+        try {
+            accessControl.checkCanInsertIntoTable(null, new SchemaTableName(schemaName, tableName));
+        }
+        catch (AccessDeniedException exception) {
+            throw new AccessDeniedException(format("Cannot create an empty partition for table %s.%s", schemaName, tableName));
```

Comment:
Use `SchemaTableName` to format the error message, as above. Maybe extract a variable for the existing `new SchemaTableName(...)`

Reply:
Actually, let's not wrap the error message here. The original message "cannot insert into table" is reasonable, and the stack trace might provide useful information. Same goes for the others.